src/strongbox/db.clj

```diff
@@ -18,7 +18,8 @@
 (defn-spec find-in-db :db/addon-catalogue-match
   "looks for `installed-addon` in the given `db`, matching `toc-key` to a `catalogue-key`.
   if a `toc-key` and `catalogue-key` are actually lists, then all the `toc-keys` must match the `catalogue-keys`"
-  [db :addon/summary-list, installed-addon :addon/installed, toc-keys :db/toc-keys catalogue-keys :db/catalogue-keys]
+  ;;[db :addon/summary-list, installed-addon :addon/installed, toc-keys :db/toc-keys catalogue-keys :db/catalogue-keys]
+  [db :addon/summary-list, installed-addon map?, toc-keys :db/toc-keys catalogue-keys :db/catalogue-keys]
```

Comment:
we've regressed from `:addon/installed` to `map`. reinstate `:addon/installed` and accommodate whatever random map data we're passing in - or specify *exactly* the keys this func and descendants need